nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/PropertiesBagResource.java

```diff
@@ -176,11 +176,17 @@ protected ActionReportResult clearThenSaveProperties(List<Map<String, String>> p
 
             for (Map<String, String> property : properties) {
                 Property existingProp = existing.get(property.get("name"));
-                String unescapredpropname = ((Object)property.get("name")).toString();
-                String escapedName = getEscapedPropertyName(unescapredpropname);
+
+                String unescapedName = ((Object)property.get("name")).toString();
+                String escapedName = getEscapedPropertyName(unescapedName);
+
                 String value = ((Object)property.get("value")).toString();
-                String description = ((Object)property.get("description")).toString();
-                final String unescapedValue = value.replaceAll("\\\\", "");
+                String unescapedValue = value.replaceAll("\\\\", "");
+                
+                String description = null;
+                if (property.get(description) != null) {
+                    description = Object.class.cast(property.get("description")).toString();
```

Comment:
That cast is super weird.  Can't you just get rid of it?

Reply:
I agree it is super weird. Annoyingly we ran into issues before where the toString() method returned a JsonString so would cause a ClassCastException. So it's irritating, but necessary